### DISCUSSION TITLE
Webcam example takes 5 seconds to get response for offer method. any solution to reduce the initial delay?

### DISCUSSION THREAD

**Discussion by yesbyen** [aiortc/aiortc#994] [Q&A]
How can reduce the initial delay for the webcam example. Now it is taking 5 to 7 seconds to load the video.

I am using the flowing configuration.

options = {"framerate": "30", "video_size": "640x480", "pixel_format":"yuyv422"}

parser.add_argument(
        "--video-codec", "-video/VP8", help="Force a specific video codec (e.g. video/H264)"
 )

**Comment by issa-ai** [comment]
Are you trying on windows ? I noticed the same thing while on windows but on ubuntu it is a lot faster.

**Comment by yesbyen** [comment]
@issa-ai  Yes, testing it on windows

**Comment by yesbyen** [comment]
Can somebody share the solution to resolve the delay in webcam (windows) sample program?

**Comment by Aroxed** [comment]
I've debugged a little the source of the library and found that the main delay happens in server.py function offer() and more precisely here:
```
    answer = await pc.createAnswer()
    await pc.setLocalDescription(answer)
``` 
looking deeper in setLocalDescription() I found these lines in .venv/Lib/site-packages/aiortc/rtcpeerconnection.py
```
        # gather candidates
        await self.__gather()
```
It takes 5-7 seconds to finish this line.
I realized that this was happening due to the presence of several IP addresses on my laptop, added by VirtualBox. I disabled all the Network Ethernet devices except my WiFi-related address, and everything started working fast!!

**Comment by Scratchcat1** [comment]
This can also occur when connection to the (default) stun server times out.

Disabling the default STUN server worked for me
```python
pc = RTCPeerConnection(RTCConfiguration(iceServers=[]))
```

By default aiortc uses a `stun.l.google.com` if `RTCConfiguration.iceServers` is `None` aren't explicitly set to `[]`.
`RTCPeerConnection()` defaults to an `RTCConfiguration` with `iceServers`  as `None`.

Inside aioice it will try to connect to the TURN and STUN servers with a default timeout of 5 seconds
https://github.com/aiortc/aioice/blob/29976f5415714df13787b933012f8a966a4859bf/src/aioice/ice.py#L972-L974
